notify/compat.go

```diff
@@ -0,0 +1,28 @@
+package notify
+
+import (
+	"encoding/json"
+
+	"github.com/grafana/alerting/definition"
+)
+
+func PostableAPIReceiverToAPIReceiver(r *definition.PostableApiReceiver) *APIReceiver {
```

Comment:
This is just moving this function and its test to notify package. No substantial changes